File: java/TaskieWithRetrofit/app/src/main/java/ada/osc/taskie/view/NewTaskActivity.java
package ada.osc.taskie.view;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.Spinner;

import ada.osc.taskie.R;
import ada.osc.taskie.model.Task;
import ada.osc.taskie.model.TaskPriority;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class NewTaskActivity extends AppCompatActivity {

	@BindView(R.id.edittext_newtask_title)	EditText mTitleEntry;
	@BindView(R.id.edittext_newtask_description) EditText mDescriptionEntry;
	@BindView(R.id.spinner_newtask_priority) Spinner mPriorityEntry;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_new_task);
		ButterKnife.bind(this);
		setUpSpinnerSource();
	}

	private void setUpSpinnerSource() {
		mPriorityEntry.setAdapter(
				new ArrayAdapter<TaskPriority>(
						this, android.R.layout.simple_list_item_1, TaskPriority.values()
				)
		);
		mPriorityEntry.setSelection(0);
	}

	@OnClick(R.id.imagebutton_newtask_savetask)
	public void saveTask(){
		String title = mTitleEntry.getText().toString();
		String description = mDescriptionEntry.getText().toString();
		TaskPriority priority = (TaskPriority) mPriorityEntry.getSelectedItem();

		Task newTask = new Task(title, description, priority);
		Intent saveTaskIntent = new Intent(this, TasksActivity.class);
		saveTaskIntent.putExtra(TasksActivity.EXTRA_TASK, newTask);
		setResult(RESULT_OK, saveTaskIntent);
		finish();
	}
}

File: kotlin/Taskie/app/src/main/java/ada/osc/taskie/view/NewTaskActivity.kt
package ada.osc.taskie.view

import ada.osc.taskie.R
import ada.osc.taskie.model.Task
import ada.osc.taskie.model.TaskPriority
import ada.osc.taskie.view.TasksActivity.Companion.EXTRA_TASK
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_new_task.*

class NewTaskActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_task)

        initUi()
    }

    private fun initUi() {
        setUpSpinnerSource()

        saveTask.setOnClickListener { saveNewTask() }
    }

    private fun saveNewTask() {
        val title = newTaskTitle.text.toString()
        val description = newTaskDescription.text.toString()
        val priority = newTaskPriority.selectedItem as? TaskPriority ?: TaskPriority.LOW

        val newTask = Task(title = title, description = description, priority = priority)

        val saveTaskIntent = Intent().apply { putExtra(EXTRA_TASK, newTask) }
        setResult(Activity.RESULT_OK, saveTaskIntent)
        finish()
    }

    private fun setUpSpinnerSource() {
        newTaskPriority.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, TaskPriority.values())

        newTaskPriority.setSelection(0)
    }
}
